cachi2/core/package_managers/gomod.py

```diff
@@ -860,12 +859,22 @@ def _resolve_gomod(
 
     config = get_config()
 
+    should_vendor = app_dir.join_within_root("vendor").path.is_dir()
+
+    if should_vendor:
+        # Even though we do not perform a "go mod download" when vendoring is detected, some
+        # go commands still download dependencies as a side effect. Since we don't want those
+        # copied to the output dir, we need to set the GOMODCACHE to a different directory.
+        gomod_cache = f"{tmp_dir}/vendor-cache"
+    else:
+        gomod_cache = f"{tmp_dir}/pkg/mod"
```

Comment:
If I'm not mistaken with my comment above, this won't be necessary and the diff is going to be smaller.